### ISSUE TITLE
mnc2nii treats minc files touched by different minc tools differently

### ISSUE THREAD

**Issue by gdevenyi** [BIC-MNI/minc-tools#69]
Running into a very weird bug in minc-toolkit/1.9.15:

Example 1:
have brain1.mnc and brain1_mask.mnc
if I do
```sh
> mincmask brain1.mnc brain1_mask.mnc brain1_extracted.mnc
> mnc2nii brain1_extracted.mnc brain1_extracted.nii
```

I get NIFTI file that has the orientation re-arranged (i.e. no longer in the same space)
If instead I do:
```sh
> mincmath -mult brain1.mnc brain1_mask.mnc brain1_extracted.mnc
> mnc2nii brain1_extracted.mnc brain1_extracted.nii
```

Everything is fine

Example 2:
have brain1_mask.mnc
```
> mnc2nii brain1_mask.mnc brain1_mask.nii
```
Everything is fine.

```
> mincmorph -successive DDDDDDDD brain1_mask.mnc brain1_mask_D.mnc
> mnc2nii brain1_mask_D.mnc brain1_mask_D.nii
```
Same problems as above.

Example of the swapped orientation problem:
![image](https://user-images.githubusercontent.com/3001850/28389313-9dde9810-6ca3-11e7-9fe1-6afa2168c94b.png)

**Comment by andrewjanke** [comment]
HI Gabriel,

My punt is a apparent vs file dimension order bug. I'm currently
having the same issues with volextents as it seems that what I
originally wrote to try to handle this doesn't work anymore:

   https://github.com/BIC-MNI/minc-widgets/blob/master/volextents/volextents#L108

I'm putting together a test case for this. I would suggest you try
this bit of magic on both the input minc files:

   mincreshape \
      -clobber \
      -2 \
      +direction \
      -dimorder zspace,yspace,xspace \
      -dimsize xspace=-1 \
      -dimsize yspace=-1 \
      -dimsize zspace=-1 \
      input.mnc output.mnc

and see if the problem still exists. If not then it's dimension
ordering and (likely) how mnc2nii detects this.


a

PS: If anyone has a more robust version of what I've used in
volextents I'm also all ears!

**Comment by rdvincent** [comment]
mnc2nii is pretty buggy. I have been working on some comprehensive fixes, but it is surprisingly hard to catch all of the cases.

**Comment by gdevenyi** [comment]
@andrewjanke sadly, the mincreshape didn't fix it up :(

**Comment by fristed** [comment]
This seems to be exactly the same bug we report in issue #32 (last comment). Also, I think this bug arose with the fix done in November 2016, which addressed the issue of wrong start values. Running a mnc2nii from before the November 2016 fix does not have the bug with swapped orientation.
In issue #32 (last comment) I provide an example, which behaves similarly as the example provided by Gabriel above. However, running an old (before Novmeber 2016) version of mnc2nii does not result in the error.

**Comment by gdevenyi** [comment]
Thanks for the hint @fristed, an older version of mnc2nii doesn't have this bug and otherwise converts everything fine.

**Comment by gdevenyi** [comment]
Fixed by #99
